Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource_printer_test.go

```diff
@@ -241,9 +246,26 @@ func TestCustomTypePrinting(t *testing.T) {
 	}
 }
 
+func TestCustomTypePrintingWithKind(t *testing.T) {
+	columns := []string{"Data"}
+	printer := NewHumanReadablePrinter(&PrintOptions{WithKind: true, Kind: "test"})
+	printer.Handler(columns, PrintCustomType)
```

Comment:
I got the feeling this test is really just testing `PrintCustomType`, which is only used in tests...

This is related to one of the first comments I've made, expressing my concern for the implementation requiring code in every print function for every type. If we forget to add that logic to one type, we'll be broken and there are no tests to capture it.